src/main/scala/scalaz/stream/wye.scala

```diff
@@ -836,9 +836,7 @@ object wye {
         }
       })(S)
 
-      repeatEval(Task.async[Seq[O]] { cb => a ! Get(cb) })
-      .flatMap(emitAll)
-      .onComplete(eval_(Task.async[Unit](cb => a ! DownDone(cb))))
+      repeatEval(Task.async[Seq[O]] { cb => a ! Get(cb) }) onHalt { _.asHalt } flatMap emitAll onComplete eval_(Task.async[Unit](cb => a ! DownDone(cb)))
```

Comment:
@djspiewak this is more about code style, but Its hard for me to read superlong scala lines, and furthemore infix syntax in scala is not that deterministics as the haskell one, so I would really be more happy with dot notation if possible?